src/function/finitediff.jl

```diff
@@ -32,5 +32,5 @@ function instantiate_function(f, x, adtype::AutoFiniteDiff, p, num_cons = 0)
         hv = f.hv
     end
 
-    return OptimizationFunction{false,AutoFiniteDiff,typeof(f),typeof(grad),typeof(hess),typeof(hv),Nothing,Nothing,Nothing}(f,adtype,grad,hess,hv,nothing,nothing,nothing)
+    return OptimizationFunction{false}(f,adtype,grad,hess,hv,nothing,nothing,nothing)
```

Comment:
why false? It's building mutating functions.

Reply:
IIUC, it is supposed to represent the behaviour of `f`, I actually spent a few minutes thinking about it since the derivatives are all in-place and confirmed from docs 

>For example, on ODEs, we have that f!(du,u,p,t) is the in-place form which, as its output, mutates du. Whatever is returned is simply ignored. Similarly, for OOP we have the form du=f(u,p,t) which uses the return.

based on which, since our objective is always `oop` I believe it would be `false`?

Reply:
The objective is not oop, it's a scalar so it's always going to be returning a value instead of mutating. So the definition here has to be dependent on the derivative functions, which are all in-place in the versions that you're currently building. There is room for alternative dispatches which do not mutate any values and instead directly compute and return a Hessian, that would be the oop dispatch.

Reply:
Oh right 🤦  
It was inconsistent between the different backends so had to be fixed